### ISSUE TITLE
A couple of issues in current state of the master branch and v1.0.0

### ISSUE THREAD

**Issue by ajdelaguila** [redhat-cop/image-scanning-signing-service#13]
Hi,

I have downloaded the project and, from the master branch, it is pointing to a non-existing version of the openshift-applier (v3.9.1) whereas the latest tag in the repository is (v2.0.8).

This is the error message that I get:
```
ansible-galaxy install -r requirements.yml -p galaxy
 [WARNING]: - openshift-applier was NOT installed successfully: - command /usr/local/bin/git checkout v3.9.1 failed in directory /Users/myusername/.ansible/tmp/ansible-local-
383701sxdUG/tmpnsOq_m/openshift-applier (rc=1)

ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.
```

When I change the tag to point to the latest tag v2.0.8 of the openshift-applier, I get this error when trying to execute the provisioning:
```
ansible-playbook -i inventory/ galaxy/openshift-applier/playbooks/openshift-cluster-seed.yml -e filter_tags=core

PLAY [seed-hosts[0]] ************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************
ok: [localhost]

TASK [openshift-applier : Check applier requirements] ***************************************************************************************************************************
included: /Users/myusername/xvprojects/image-scanning-signing-service/galaxy/openshift-applier/roles/openshift-applier/tasks/pre-check.yml for localhost

TASK [openshift-applier : Exit if ansible version doesn't meet minimum requirements] ********************************************************************************************
skipping: [localhost]

TASK [openshift-applier : Retrieve oc client version] ***************************************************************************************************************************
changed: [localhost]

TASK [openshift-applier : Filter out just the oc version number] ****************************************************************************************************************
ok: [localhost]

TASK [openshift-applier : Do *not* use the 'ignore_unknown_parameters' flag if 'oc' version is older than 3.7] ******************************************************************
skipping: [localhost]

TASK [openshift-applier : Error out on anything in the inventory that is no longer supported] ***********************************************************************************
included: /Users/myusername/xvprojects/image-scanning-signing-service/galaxy/openshift-applier/roles/openshift-applier/tasks/error-on-unsupported.yml for localhost

TASK [openshift-applier : Look for unsupported top-level dictionary entries] ****************************************************************************************************
skipping: [localhost] => (item=content_dir)

TASK [openshift-applier : Look for unsupported content-level dictionary entries] ************************************************************************************************
skipping: [localhost] => (item=[{u'content': [{u'name': u'imagesigningrequest', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../examples/image-signing-request-crd.yml', u'tags': [u'core']}, {u'name': u'imagescanningrequest', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../examples/image-scanning-request-crd.yml', u'tags': [u'core']}], u'object': u'crd'}, u'file_action'])
skipping: [localhost] => (item=[{u'content': [{u'name': u'imagesigningrequest', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../examples/image-signing-request-crd.yml', u'tags': [u'core']}, {u'name': u'imagescanningrequest', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../examples/image-scanning-request-crd.yml', u'tags': [u'core']}], u'object': u'crd'}, u'template_action'])
ok: [localhost] => (item=[{u'content': [{u'file_action': u'create', u'name': u'image-management-project', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/projects/projects-image-management.yml', u'tags': [u'core']}, {u'file_action': u'create', u'name': u'image-scanning-signing-project', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/projects/projects-image-scanning-signing-pipeline.yml', u'tags': [u'pipeline']}], u'object': u'projectrequest'}, u'file_action'])
skipping: [localhost] => (item=[{u'content': [{u'file_action': u'create', u'name': u'image-management-project', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/projects/projects-image-management.yml', u'tags': [u'core']}, {u'file_action': u'create', u'name': u'image-scanning-signing-project', u'file': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/projects/projects-image-scanning-signing-pipeline.yml', u'tags': [u'pipeline']}], u'object': u'projectrequest'}, u'template_action'])
skipping: [localhost] => (item=[{u'content': [{u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/core/params', u'name': u'core-policy', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/core/template.yml', u'tags': [u'core']}, {u'namespace': u'image-scanning-signing-pipeline', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/image-scanning-signing/params-scanning-signing-pipeline', u'name': u'image-scanning-signing-application-policy', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/image-scanning-signing/role-binding-template.yml', u'tags': [u'pipeline']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/scc/params', u'name': u'privileged-scc', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/scc/template.yml', u'tags': [u'core']}], u'object': u'policy'}, u'file_action'])
skipping: [localhost] => (item=[{u'content': [{u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/core/params', u'name': u'core-policy', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/core/template.yml', u'tags': [u'core']}, {u'namespace': u'image-scanning-signing-pipeline', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/image-scanning-signing/params-scanning-signing-pipeline', u'name': u'image-scanning-signing-application-policy', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/policy/image-scanning-signing/role-binding-template.yml', u'tags': [u'pipeline']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/scc/params', u'name': u'privileged-scc', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/scc/template.yml', u'tags': [u'core']}], u'object': u'policy'}, u'template_action'])
skipping: [localhost] => (item=[{u'content': [{u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/gpg/params', u'namespace': u'image-management', u'name': u'gpg', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/gpg/template.yml', u'tags': [u'core']}], u'object': u'gpg'}, u'file_action'])
skipping: [localhost] => (item=[{u'content': [{u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/gpg/params', u'namespace': u'image-management', u'name': u'gpg', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/gpg/template.yml', u'tags': [u'core']}], u'object': u'gpg'}, u'template_action'])
skipping: [localhost] => (item=[{u'content': [{u'namespace': u'image-management', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/imagestreams/params', u'name': u'images', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/imagestreams/images.yml', u'tags': [u'core']}], u'object': u'imagestreams'}, u'file_action'])
skipping: [localhost] => (item=[{u'content': [{u'namespace': u'image-management', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/imagestreams/params', u'name': u'images', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/imagestreams/images.yml', u'tags': [u'core']}], u'object': u'imagestreams'}, u'template_action'])
skipping: [localhost] => (item=[{u'content': [{u'namespace': u'image-scanning-signing-pipeline', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-pipeline/params', u'name': u'image-scanning-signing-application', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-pipeline/template.yml', u'tags': [u'pipeline']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/s2i-golang/params', u'namespace': u'image-management', u'name': u's2i-golang', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/s2i-golang/template.yml', u'tags': [u'core']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-sign-scan-base/params', u'namespace': u'image-management', u'name': u'image-sign-scan-base', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-sign-scan-base/template.yml', u'tags': [u'core']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-service/params', u'namespace': u'image-management', u'name': u'image-scanning-signing-service', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-service/template.yml', u'tags': [u'core']}], u'object': u'builds'}, u'file_action'])
skipping: [localhost] => (item=[{u'content': [{u'namespace': u'image-scanning-signing-pipeline', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-pipeline/params', u'name': u'image-scanning-signing-application', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-pipeline/template.yml', u'tags': [u'pipeline']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/s2i-golang/params', u'namespace': u'image-management', u'name': u's2i-golang', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/s2i-golang/template.yml', u'tags': [u'core']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-sign-scan-base/params', u'namespace': u'image-management', u'name': u'image-sign-scan-base', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-sign-scan-base/template.yml', u'tags': [u'core']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-service/params', u'namespace': u'image-management', u'name': u'image-scanning-signing-service', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/builds/image-scanning-signing-service/template.yml', u'tags': [u'core']}], u'object': u'builds'}, u'template_action'])
skipping: [localhost] => (item=[{u'content': [{u'namespace': u'image-scanning-signing-pipeline', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-pipeline/params-jenkins', u'name': u'deploy jenkins', u'template': u'openshift//jenkins-ephemeral', u'tags': [u'pipeline']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-service/params', u'namespace': u'image-management', u'name': u'image-scanning-signing-service', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-service/template.yml', u'tags': [u'core']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-pipeline/params', u'namespace': u'image-scanning-signing-pipeline', u'name': u'image-scanning-signing-application', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-pipeline/template.yml', u'tags': [u'pipeline']}], u'object': u'deployments'}, u'file_action'])
skipping: [localhost] => (item=[{u'content': [{u'namespace': u'image-scanning-signing-pipeline', u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-pipeline/params-jenkins', u'name': u'deploy jenkins', u'template': u'openshift//jenkins-ephemeral', u'tags': [u'pipeline']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-service/params', u'namespace': u'image-management', u'name': u'image-scanning-signing-service', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-service/template.yml', u'tags': [u'core']}, {u'params': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-pipeline/params', u'namespace': u'image-scanning-signing-pipeline', u'name': u'image-scanning-signing-application', u'template': u'/Users/myusername/xvprojects/image-scanning-signing-service/inventory/../files/deployments/image-scanning-signing-pipeline/template.yml', u'tags': [u'pipeline']}], u'object': u'deployments'}, u'template_action'])

TASK [openshift-applier : Error out if any unsupported items were found] ********************************************************************************************************
failed: [localhost] (item={'value': 2, 'key': u'file_action'}) => {"changed": false, "item": {"key": "file_action", "value": 2}, "msg": "One or more unsupported inventory entry found for the following key: file_action"}
	to retry, use: --limit @/Users/myusername/xvprojects/image-scanning-signing-service/galaxy/openshift-applier/playbooks/openshift-cluster-seed.retry

PLAY RECAP **********************************************************************************************************************************************************************
localhost                  : ok=6    changed=1    unreachable=0    failed=1
```
 
Taking a look at the inventory file, I have only found two occurrences of "file_action" and they are both set to "create".

I have also switched to the version tagged as v1.0.0 and it is also pointing to the same version of the openshift-applier and after pointing to the latest one I get the same error.

Let me know if there is any other information I can provide.

**Comment by sabre1041** [comment]
@ajdelaguila pushed a change. please verify

**Comment by sabre1041** [comment]
closed via 9393afa8bebcd5390b9d0a45d7be6ffd1b3880b9
